File: algorithms/scala/Factorial.scala

/**
 * Find the factorial (n!) of a number (n), in this case 5.
 */
object Factorial extends Application {

  println( factorial( 5 ) )

  /**
   * Recursively compute the factorial of a number.
   *
   * @param number the number to compute for.
   * @return the computed factorial of specified number.
   */
  def factorial( number: Long ): Long = {

    number match {

      case 1 => 1
      case _ => number * factorial( number - 1 )

    }

  }

}

File: algorithms/ruby/factorial.rb
module Sequences

  public

  # Recursively compute the factorial of a number.
  #
  def factorial( num )

    return 1 if num <= 1
    num * factorial( num - 1 )

  end

  private

end

if __FILE__ == $0

  include Sequences

  # Find the factorial (n!) of a number (n), in this case 5.

  puts factorial 1
  puts factorial 2
  puts factorial 3
  puts factorial 4
  puts factorial 5

end